src/charm_state.py

```diff
@@ -99,7 +127,7 @@ class State:
         jenkins_agent_service_name: The Jenkins agent workload container name.
     """
 
-    agent_meta: metadata.Agent
+    agent_meta: AgentMetadata
```

Comment:
Minor nitpick: if the attribute is called `agent_meta` why not call the class `AgentMeta`? Or vice-verse?

Reply:
good point, I updated the class name